src/peer/identity.js

```diff
@@ -39,12 +40,17 @@ function loadOrGenerateIdentity(filePath: string): Promise<PeerId> {
       })
 }
 
-function inflateMultiaddr(multiaddrString: string): Promise<PeerInfo> {
-    const multiaddr = Multiaddr(multiaddrString)
-    return PeerInfo.create().then(peerInfo => {
-      peerInfo.multiaddr.add(multiaddr)
-      return peerInfo
-    })
+function inflateMultiaddr(multiaddrString: string): PeerInfo {
+  const multiaddr = Multiaddr(multiaddrString)
+  const ipfsIdB58String = multiaddr.stringTuples().filter((tuple) => {
```

Comment:
One small thing: we should throw with a useful error if there is no matching tuple